internal/pkg/v2/infrastructure/redis/client.go

```diff
@@ -294,6 +296,20 @@ func (c *Client) AddDevice(d model.Device) (model.Device, errors.EdgeX) {
 	return addDevice(conn, d)
 }
 
+// Delete all readings and events that have been pushed (pushed timestamp is greater than zero)
+func (c *Client) DeletePushedEvents() errors.EdgeX {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	pushedEventIds, readingIds, err := getPushedEventReadingIds(conn)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	deleteReadingsByIdChannel <- readingIds
+	deleteEventsByIdChannel <- pushedEventIds
+	return nil
```

Comment:
This logic should be in Core Data's app layer. DB layer expose GetPushedEventReadingIds

Reply:
Moved to the app layer under DeletePushedEvents func.